tests/end_to_end/server_test.py

```diff
@@ -53,3 +53,34 @@ def runClientCmd(self, client, command):
         except subprocess.CalledProcessError as error:
             self.server.printDebugInfo()
             raise error
+
+
+class RemoteServerTest(ServerTest):
+    """
+    A test that uses both the ga4gh server and a remote server
+    that serves data files
+    """
+    def otherSetup(self):
+        remoteDataDir = self.getRemoteDataDir()
+        self.remoteServer = server.RemoteServerForTesting(remoteDataDir)
+        self.remoteServer.start()
+
+    def otherTeardown(self):
+        self.remoteServer.shutdown()
+
+    def otherExceptionHandling(self):
+        self.remoteServer.printDebugInfo()
+
+    def getRemoteDataDir(self):
+        raise Exception("Must subclass RemoteServerTest")
```

Comment:
I think `NotImplementedException` is the exception used to indicate an abstract method being called directly.